src/tests/ftest/util/dmg_utils.py

```diff
@@ -27,6 +27,7 @@
 from grp import getgrgid
 from pwd import getpwuid
 import re
+from collections import defaultdict
```

Comment:
This brings up an interesting discussion.  If we don't have all of the expected data, won't returning an empty dictionary be sufficient or even preferable?  It would more clearly indicate something was wrong with the command output processing (`DmgCommand.result.exit_status == 0 and <return_value> is None`).

Reply:
I think you have a good point. The regex I wrote only works with correctly formed output. If it's not, for example no hardware storage, then it returns a partially filled dictionary. At this point, I would say I provide a utility function that covers general case only because the use cases aren't clear.